src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -576,6 +576,29 @@ export const selectPortfolioAccountBalanceByAccountNumberAndChainId = createCach
     `${filter?.accountNumber}-${filter?.chainId}` ?? 'accountNumber-chainId',
 )
 
+export type PortfolioAccountIdByNumberByChainId = {
+  [accountNumber: number]: { [chainId: ChainId]: AccountId }
+}
+
+export const selectPortfolioAccountIdByNumberByChainId = createDeepEqualOutputSelector(
+  selectPortfolioAccountsFiatBalancesIncludingStaking,
```

Comment:
Seems hacky to use a selector that returns fiat balances to get AccountIds, we should be able to use `selectPortfolioAccountMetadata` / `selectPortfolioAccounts` ?

Reply:
This doesnt return fiat balances, it returns a lookup `accountNumber => chainId => accountId`

Those other selectors are indexed by accountId to return account metadata, but for this application we need a selector that allows constant time lookup of the accountId using accountNumber and chainId, i.e we need to find the accountId rather than use it to find other information

Reply:
perhaps i should rename it to something like `selectPortfolioAccountIdLookupByNumberByChainId` ?

Reply:
I initially misread this comment. Yes great idea.